### DISCUSSION TITLE
Parsing `extra` again without `remove` + `require`

### DISCUSSION THREAD

**Discussion by bilogic** [composer/composer#10522] [Q&A / Support]
Hi,

1. I symlink some packages of mine into a project's `vendor` folder
2. I find that editing my package's `composer.json`'s `extra` section has no effect, i.e. `composer update` does not seem to recognize those new bits
3. In the end, I have to `remove`, followed by `require` the package again
4. Is there a way/command to parse the `extra` in `composer.json` again?
5. Perhaps the bigger question here is, what is the difference between `update` vs `remove + require`?

See the actual use case here https://github.com/barryvdh/laravel-ide-helper/issues/1282

Thank you!

**Comment by Seldaek** [comment]
If you want to symlink a package, you should either use https://getcomposer.org/doc/05-repositories.md#path or https://github.com/franzliedke/studio to achieve that. Doing it guerilla style means that updates do not update the package metadata, and thus things like this do not work. The composer.json inside vendor/foo/bar/ is never actually read. Metadata is stored in vendor/composer/installed.json and comes from packagist.org or other package repositories you declare, it's not read from disk (unless you use a path repo which fixes that).

**Reply by bilogic** [reply]
Got it thanks!

**Reply by bilogic** [reply]
@Seldaek 

Clarification: actually, I'm using `composer.json`'s `repositories`, that's why `remove` + `require` in my step 3 works. 

The symlink into `vendor` is performed by composer. I think this is what you call a path repo, correct?

In this case, is `composer update` supposed to re-read the `composer.json` in vendor/foo/bar (including `extra`)?

**Reply by Seldaek** [reply]
Well I don't know what the `extra` is supposed to do in your case, so I don't know what the problem is. Maybe whatever reads it once does not re-read it properly when it updates using a path repo. There are not enough details here to help you further..

**Reply by bilogic** [reply]
@Seldaek 

I assumed all along that handling `extra` was a composer behavior, but I just googled and saw that `extra` is processed by `scripts`, the specifics of my issue can be found here https://github.com/barryvdh/laravel-ide-helper/issues/1282

**Reply by Seldaek** [reply]
Yeah except I have no idea how this whole laravel auto discovery stuff works and I'm not so inclined to dig into it without an easy way to reproduce the problem.

**Reply by bilogic** [reply]
It's ok, I understand. If you don't mind, do you know if `composer update` copies an existing path repo's modified `extra` into  `installed.json`?
